src/zos-tls.cc

```diff
@@ -96,3 +92,24 @@ void __tlsvaranchor_destroy(struct __tlsanchor *anchor) {
 void *__tlsPtrFromAnchor(struct __tlsanchor *anchor, const void *initvalue) {
   return __tlsPtrAlloc(anchor->sz, &(anchor->key), &(anchor->once), initvalue);
 }
+void * __tlsValue(tls_t *a) {
+	char * initvalue = NULL;
+	void *val = NULL;
+	initvalue = (char *)malloc(sizeof(char)*(a->sz));
+	assert(initvalue != NULL);
+	bzero(initvalue,a->sz);
+	val = __tlsPtrAlloc(a->sz, &(a->key), &(a->once),(void *)initvalue);
+	free(initvalue);
+	return val;
```

Comment:
Suggest storing the memory malloc'd by `__tlsPtrAlloc` (`val - sizeof(pthread_key_t)`) in `tls_t`, so that `__tlsvaranchor_destroy` can then free it.

Reply:
Please replace all new tabs by 2 single blanks to be consistent with the other code.

Reply:
Replaced all tabs with 2 blanks. The malloc inside __tlsPtrAlloc is existing code and gets freed in __tlsvaranchor_destroy.

Reply:
@IgorTodorovskiIBM can you please approve this PR?

Reply:
> The malloc inside __tlsPtrAlloc is existing code and gets freed in __tlsvaranchor_destroy.

True, the malloc is existing code, but it doesn't get freed:
```
void __tlsvaranchor_destroy(struct __tlsanchor *anchor) {
  pthread_key_delete(anchor->key);
  free(anchor);
}
```
I can fix this in a separate PR.

Reply:
@gabylb It is not cleaned up in the destructor function?

Reply:
You are right. I missed the call to [pthread_setspecific(*k, p)](https://github.com/ibmruntimes/zoslib/blob/865e854eff1810c7459e566e7da5190c786b31cc/src/zos-tls.cc#L66) after p is malloc'd.